vllm/model_executor/models/phi3v.py

```diff
@@ -418,39 +431,55 @@ def input_processor_for_phi3v(ctx: InputContext, llm_inputs: LLMInputs):
             logger.warning("Please follow the prompt format that is "
                            "documented on HuggingFace which does not involve "
                            "repeating <|image|> tokens.")
-        elif len(re.findall(r"(<\|image_\d+\|>)+", prompt)) > 1:
-            logger.warning("Multiple image input is not supported yet, "
-                           "so any extra image tokens will be treated "
-                           "as plain text.")
-
+        elif (num_image_tags := len(re.findall(r"(<\|image_\d+\|>)+",
```

Comment:
I would try to only call `re.findall` before the `if` statement so it doesn't have to be computed again later in line 441.